jwst/datamodels/slit.py

```diff
@@ -155,10 +170,23 @@ def __init__(self, init=None, data=None, dq=None, err=None,
 
         if var_rnoise is not None:
             self.var_rnoise = var_rnoise
-        # Implicitly create arrays
-        self.dq = self.dq
-        self.err = self.err
-
+        
+        if dq is not None:
+            self.dq = dq
+        if err is not None:
+            self.err = err
+        if bunit_data is not None:
+            self.meta.bunit_data = bunit_data
+        if bunit_err is not None:
+            self.meta.bunit_err = bunit_err
```

Comment:
Since when is it necessary to populate meta attributes in datamodel class?

Reply:
It's not necessary in general. It only covers the case when a model is initialized not from a file but passing in the (items, values) specified in the schema. 
When the data arrays are extracted, a SlitDataModel is initialized by passing the arrays to the initializer. The meta data don't matter in this case as they are stored in MulitSlitModel.meta. But when it's written to a fits file the extensions need to have this kw in the header. That's why it's added to the initializer. This is all used internally in extract_2d.

Reply:
In the future, my goal is to completely strip all `meta` items from `SlitDataModel`, including the wcs and keep everything in MultiSlitModel.meta. Then when a SlitModel is created from a SlitDataModel the meta and the wcs will be populated. But let's get first this working again.